image-customize

```diff
@@ -272,10 +276,17 @@ def main():
     if '/' not in args.base_image:
         subprocess.check_call([os.path.join(BOTS_DIR, "image-download"), args.base_image])
     network = testvm.VirtNetwork(0, image=args.base_image)
-    machine = testvm.VirtMachine(maintain=True,
-                                 verbose=args.verbose,
-                                 networking=network.host(restrict=args.no_network),
-                                 image=prepare_install_image(args.base_image, args.image, args.resize, args.fresh))
+    machine_args = {
+        'maintain': True,
+        'verbose': args.verbose,
+        'networking': network.host(restrict=args.no_network),
+        'image': prepare_install_image(args.base_image, args.image, args.resize, args.fresh),
+    }
+    if args.cpus:
+        machine_args['cpus'] = args.cpus
+    if args.memory_mb:
+        machine_args['memory_mb'] = args.memory_mb
+    machine = testvm.VirtMachine(machine_args)
```

Comment:
This looks really weird. Instead of passing kwargs you now pass a single positional argument of a dictionary type. I would understand it if it would be `VirtMachine(**machine_args)`.

However, this indirection is completely unnecessary. Both the `args.*` values and the kwargs of VirtMachine default to `None`, so you can just pass them as they are. I.e.

    VirtMachine(maintain=True, ..., cpus=args.cpus, memory_mb=args.memory_mb)

Reply:
> I would understand it if it would be `VirtMachine(**machine_args)`.

Yes, that was it. That's what I get by applying wrong patches that late in the evening ...